### DISCUSSION TITLE
How can I reference images outside sourceCodeDir?

### DISCUSSION THREAD

**Discussion by adamreisnz** [ElMassimo/vite_ruby#460] [Q&A]
Hello, I figured I'd open a new question as the old item is already marked as answered, so not getting a lot of traction.

Could someone please provide a concrete working example for how to get images working from the `/app/assets/images` folder when the `sourceCodeDir` is `/app/javascript`, in a Rails project?

I have tried a lot of different things, but can't seem to get it to work unless I move the images into the `/app/javascript` folder.

**vite.json**
```json
{
  "all": {
    "sourceCodeDir": "app/javascript",
    "additionalEntryPoints": [
      "assets/images/**/*"
    ]
  },
  "development": {
    "autoBuild": true,
    "publicOutputDir": "vite-dev",
    "port": 3036
  },
  "test": {
    "autoBuild": true,
    "publicOutputDir": "vite-test",
    "port": 3037
  }
}

```
**Note:** I have tried this with or without the `additionalEntryPoints`

**Image tags**
```
<%= vite_image_tag 'images/test.svg' %>
```
This is not working

**Direct import**
```
import testImage from 'images/test.svg'
```
This is not working, unless using a resolve alias

Any suggestions would be appreciated!

**Comment by ElMassimo** [comment]
> move the images into the /app/javascript folder

That's the recommended approach.

> can't seem to get it to work

It's an unusual configuration, but as [mentioned in `additionalEntrypoints`](https://vite-ruby.netlify.app/config/index.html#additionalentrypoints) and [in the _Advanced Usage_ section](https://vite-ruby.netlify.app/guide/advanced.html#one-entrypoint-per-component), if you needed to reference assets __outside__ the `sourceCodeDir`, you need to disambiguate by starting with a leading slash:

```ruby
<%= vite_image_tag '/assets/images/test.svg' %>
```

Otherwise, it will attempt to [resolve to the `entrypointsDir` or the `sourceCodeDir`](https://vite-ruby.netlify.app/guide/rails.html#tag-helpers-%F0%9F%8F%B7):
<img width="794" alt="Screenshot 2024-06-04 at 09 51 18" src="https://github.com/ElMassimo/vite_ruby/assets/1158253/d32897b8-9f5a-4fe7-985c-ca30a591ce8c">

Regarding JS imports, read about [_Import Aliases_](https://vite-ruby.netlify.app/guide/development.html#import-aliases-%F0%9F%91%89) and the example in [`watchAdditionalPaths`](https://vite-ruby.netlify.app/config/#watchadditionalpaths).

**Comment by adamreisnz** [comment]
Regretfully, this just doesn't work. It makes no difference if I refer to the images with a leading slash or not. The images just aren't loaded by `vite_image_tag` unless they are placed in the `/app/javascript` folder.

I have read all the documentation multiple times, and tried everything I could think of. I'll just resort to moving all the images.

**Comment by ravensnowbird** [comment]
Quick fix that worked for me is dump all the images inside `app/javascript/entrypoints` folder and use the method `vite_asset_path`
